libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
We could at least limit it to Enterprise by putting this inside a `ENTERPRISE_FUNC` macro. This detection runs very early, so it should be sufficient.